e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/generate-fixed-fee-adjustment-with-amendments.spec.js

```diff
@@ -0,0 +1,123 @@
+    .withPayments([matchingPayment])
+    .build();
+
+  const tfmFacility = {
+    facilitySnapshot: {
+      coverStartDate: new Date('2023-01-01').getTime(),
+      // 1095 days after report period end
+      coverEndDate: new Date('2027-03-01').getTime(),
+      interestPercentage: 100,
+      dayCountBasis: 1,
+      value: 700000,
+      coverPercentage: 100,
+      ukefFacilityId: facilityId,
+    },
+    amendments: [
+      {
+        status: AMENDMENT_STATUS.COMPLETED,
+        effectiveDate: dateAfterReportPeriodEnd.getTime(),
+        // 365 days after report period end
+        coverEndDate: new Date('2025-03-01').getTime(),
+      },
+      {
+        status: AMENDMENT_STATUS.IN_PROGRESS,
+        effectiveDate: dateBeforeReportPeriodEnd.getTime(),
+        // 370 days after report period end
+        coverEndDate: new Date('2025-03-06').getTime(),
+      },
+      {
+        status: AMENDMENT_STATUS.COMPLETED,
+        effectiveDate: dateBeforeDateBeforeReportPeriodEnd.getTime(),
+        // 730 days after report period end
+        coverEndDate: new Date('2026-03-01').getTime(),
+      },
+    ],
+  };
+
+  beforeEach(() => {
+    cy.task(NODE_TASKS.DELETE_ALL_TFM_FACILITIES_FROM_DB);
+    cy.task(NODE_TASKS.INSERT_TFM_FACILITIES_INTO_DB, [tfmFacility]);
+
+    cy.task(NODE_TASKS.DELETE_ALL_FROM_SQL_DB);
+
+    cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [report]);
+    cy.task(NODE_TASKS.INSERT_FEE_RECORDS_INTO_DB, [feeRecord]);
+
+    pages.landingPage.visit();
+    cy.login(USERS.PDC_RECONCILE);
+
+    cy.visit(`utilisation-reports/${reportId}`);
+  });
+
+  it('should calculate the fixed fee adjustment using effective amendment at report period end', () => {
+    pages.utilisationReportPage.premiumPaymentsTab.generateKeyingDataButton().click();
+    pages.checkKeyingDataPage.generateKeyingSheetDataButton().click();
+
+    cy.url().should('eq', relative(`/utilisation-reports/${reportId}#keying-sheet`));
+
+    /**
+     * The fixed fee adjustment is the difference between
+     * the current fixed fee and the previous fixed fee,
+     * where the previous fixed fee is equal to the value
+     * stored on the FacilityUtilisationData table, and the
+     * current fixed fee is given by the product of the below
+     * values:
+     * - utilisation: 1
+     * - coverPercentage: 100 / 100 = 1
+     * - bank margin: 0.9 (this is a fixed, constant value)
+     * - interest percentage: 100 / 100 = 1
+     * - number of days left in cover period divided by day count basis: 730 / 1 = 730
+     * This yields a current utilisation value of 730 * 0.9 = 657
+     *
+     * The adjustment is therefore the difference between the previous
+     * fixed fee which was 700 and 657 which is a decrease of 43
+     */
+    cy.assertText(pages.utilisationReportPage.keyingSheetTab.fixedFeeAdjustmentDecrease(feeRecordId), '43.00');
+    cy.assertText(pages.utilisationReportPage.keyingSheetTab.fixedFeeAdjustmentIncrease(feeRecordId), '-');
```

Comment:
Would be nice to destructure the `pages` in this test so it reads nicer without losing context e.g

```js
utilisationReportPage.premiumPaymentsTab.generateKeyingDataButton().click();
checkKeyingDataPage.generateKeyingSheetDataButton().click();

cy.assertText(keyingSheetTab.fixedFeeAdjustmentDecrease(feeRecordId), '43.00');
cy.assertText(keyingSheetTab.fixedFeeAdjustmentIncrease(feeRecordId), '-');
```

Not a blocker, seems the majority of tests don't do this